FUNDAMENTALS.md

```diff
@@ -0,0 +1,133 @@
+This document covers fundamentals of Flutter cocoon, including frontend [dashboard](https://flutter-dashboard.appspot.com/) and backend APIs/Cronjobs.
+
+# Terminology
+
+* DeviceLab
+  * The hardware lab
+* Dashboard
+  * The [web UI](https://flutter-dashboard.appspot.com) that aggregates Flutter CI build results and benchmarks
+* Agent
+  * One of the real/virtual machines that runs build tasks. Typically an agent  in the devicelab will have a mobile device attached to it.
+* CLI
+  * One command line interface accessible on the dashboard
+* Cronjob
+  * Cronjob implemented in [app engine](https://pantheon.corp.google.com/appengine/cronjobs?project=flutter-dashboard)
+
+# Backend
+## APIs
+### /api/authorize-agent
```

Comment:
I think these technical details of the endpoints should be documented in the code instead. For instance, adding this as class level documentation to `lib/src/request_handlers/authorize_agent.dart` (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#leave-breadcrumbs-in-the-comments)

Reply:
I was thinking about to provide a central place introducing all APIs' functionality. As @godofredoc suggested, we will move the doc to g3. I will think about how to present the doc. considering your comment. Thanks.

Closing this PR here.